### DISCUSSION TITLE
Memory leak using emitter and program fluid

### DISCUSSION THREAD

**Discussion by BKCollaboration** [openscenegraph/OpenSceneGraph#1268] [Q&A]
Hello,

I have a lot of objects, created at run-time, loading some osgParticle setup using an emitter and a program fluid added to a Transform with addChild.

```
m_obj->addChild(...some node1)
m_obj->addChild(...some node2)
m_obj->addChild(m_emitter);
m_obj->addChild(m_program_fluid);
```

I need to remove the emitter and the program fluid at some time and, to do that, I call a removeChild in a callback. Normally, with others type of node I have no problems and the related node is flushed away. With the emitter and program fluid, the removeChild fails returning false. For example, I can see this track where I'm removing some children from a node: 

```
removing Component_0_Shoot_1_BDOF_G (Group) from a node with num children 84, child index to remove 81, result 1.
removing Component_0_Shoot_1_AA_G as (Group) from a node with num children 83, child index to remove 81, result 1.
removing Component_0_Shoot_1_SPHERE_G (Group) from a node with num children 82, child index to remove 81, result 1.
removing Splash_1_PSU (ParticleSystemUpdater) from a node with num children 119, child index to remove 117, result 1.
removing Splash_1_GEOPS (Geode) from a node with num children 118, child index to remove 117, result 1.
removing Splash_1_EMITTER (ModularEmitter) from a node with num children 117, child index to remove 117, result 0.
removing  Splash_1_FP (FluidProgram) from a node with num children 117, child index to remove 117, result 0.
```

Basically, children like Group, Geode, ParicleSystemUpdater are removed correctly but the ModularEmitter and FluidProgram return 0 because the index and num children are the same value (117).

Can you give me some ideas about the problem? I think the problem is the incorrect child index of ModularEmitter and FluidProgram, but right now I'm not able to figure out why it's happens. Changing the order of "addChild" the result is the same.

Many thanks,
bye
